src/herder/HerderImpl.cpp

```diff
@@ -892,6 +892,116 @@ HerderImpl::getJsonQuorumInfo(NodeID const& id, bool summary, bool fullKeys,
     return ret;
 }
 
+Json::Value
+HerderImpl::getJsonTransitiveQuorumInfo(NodeID const& rootID, bool summary,
+                                        bool fullKeys)
+{
+    Json::Value ret;
+    Json::Value& nodes = ret["nodes"];
+
+    auto& q = mPendingEnvelopes.getCurrentlyTrackedQuorum();
+
+    auto rootLatest = getSCP().getLatestMessage(rootID);
+    std::map<Value, int> knownValues;
+
+    // walk the quorum graph, starting at id
+    std::unordered_set<NodeID> visited;
+    std::deque<NodeID> next;
+    next.push_back(rootID);
+    visited.emplace(rootID);
+    int distance = 0;
+    int valGenID = 0;
+    while (!next.empty())
+    {
+        std::deque<NodeID> backlog(std::move(next));
+        next.clear();
+        std::sort(backlog.begin(), backlog.end());
+        for (auto const& id : backlog)
+        {
+            Json::Value cur;
+            valGenID++;
+            cur["node"] = fullKeys ? mApp.getConfig().toStrKey(id)
+                                   : mApp.getConfig().toShortString(id);
+            if (!summary)
+            {
+                cur["distance"] = distance;
+            }
+            auto it = q.find(id);
+            std::string status;
+            if (it != q.end())
+            {
+                auto qSet = it->second;
+                if (qSet)
+                {
+                    if (!summary)
+                    {
+                        cur["qset"] =
+                            getSCP().getLocalNode()->toJson(*qSet, fullKeys);
+                    }
+                    LocalNode::forAllNodes(*qSet, [&](NodeID const& n) {
+                        auto b = visited.emplace(n);
+                        if (b.second)
+                        {
+                            next.emplace_back(n);
+                        }
+                    });
+                }
+                auto latest = getSCP().getLatestMessage(id);
+                if (latest)
+                {
+                    auto vals = Slot::getStatementValues(latest->statement);
+                    // updates the `knownValues` map, and generate a unique ID
+                    // for the value (heuristic to group votes)
+                    int trackingValID = -1;
```

Comment:
I feel like this use of a numeric tracking-value-ID is more confusing than it's worth: the user will have to remember that this isn't _really_ the value voted on, but a proxy for it. I think it would be better to emit the hash here. If it really helps you eyeball problems, maybe emit both.

Reply:
done